### PR TITLE
ci: add pre-commit hooks

### PR DESCRIPTION
**Pull request by isaranto**
- Adding checks in file .pre-commit-config.yaml:
    - check-yaml
    - end-of-file-fixer
    - trailing-whitespace
    - [isort](https://github.com/pycqa/isort)
    - [ruff](https://github.com/charliermarsh/ruff-pre-commit)
    - [black](https://github.com/psf/black)
    - [pyupgrade](https://github.com/asottile/pyupgrade)

- Add a Github Action that will run this test using a [predefined action](https://github.com/pre-commit/action). This is a required check for Pull Requests.
- Includes the changes required by pre-commit in order for the tests to pass

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
```
Comment (kevinbazira) [inline .github/workflows/ci.yml]: A deprecation warning is thrown after the pre-commit job has run:
https://github.com/wikimedia/liftwing-python/actions/runs/9481533171

I looked at the affected actions and they have new versions we could use:
1. checkout: https://github.com/actions/checkout?tab=readme-ov-file#usage
2. setup-python: https://github.com/actions/setup-python?tab=readme-ov-file#basic-usage

File: liftwing/models/liftwing_model.py [hunk 2]
```diff
@@ -10,7 +10,12 @@ def __init__(self, base_url: str, payload_model: Type[BaseModel] = None):
...
 
-    def request(self, payload: Dict[str, Any], method: str = "POST", headers: Dict[str, str] = None) -> Dict[str, Any]:
+    def request(
+        self,
+        payload: Dict[str, Any],
+        method: str = "POST",
+        headers: Dict[str, str] = None,
+    ) -> Dict[str, Any]:
         """
         This is an abstact method that makes a request to the API endpoint.
```
Comment (kevinbazira) [inline liftwing/models/liftwing_model.py]:
```suggestion
        This is an abstract method that makes a request to the API endpoint.
```

**Review by kevinbazira — COMMENTED** [review]
Thank you for working on this Ilias. Most of it LGTM. I've made a few suggestions regarding a deprecation warning and fixing a typo.

File: liftwing/models/liftwing_model.py (hunk 2 above)
Comment (isaranto) [reply liftwing/models/liftwing_model.py]: Removed it completely as it is no longer an abstract function.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (isaranto) [reply .github/workflows/ci.yml]: Nice catch! I have updated the versions and also added python version >= 3.9 as required.

**Review by kevinbazira — APPROVED** [review]
LGMT!

**Pull request merged**